packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/string_database/index_db/indexed_db_adapter.dart

```diff
@@ -0,0 +1,149 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+import 'dart:js_util';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/string_database.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/string_database/string_database_element.dart';
+import 'package:amplify_core/amplify_core.dart';
+// ignore: implementation_imports
+import 'package:aws_common/src/js/indexed_db.dart';
+
+// TODO(kylechen): Consider merging/refactoring with existing 'amplify_secure_storage_web - _IndexedDBStorage' class
+/// {@template amplify_analytics_pinpoint_dart.indexed_db_adapter}
+/// Provide indexDB methods for storing/retrieving Strings
+/// {@endtemplate}
+class IndexedDbAdapter implements StringDatabase {
+  /// {@macro amplify_analytics_pinpoint_dart.indexed_db_adapter}
+  IndexedDbAdapter();
+
+  /// The name of the database
+  ///
+  /// Reference: https://www.w3.org/TR/IndexedDB/#name
+  String get databaseName => 'amplify_analytics_pinpoint';
+
+  /// The name of the object store
+  ///
+  /// Reference: https://www.w3.org/TR/IndexedDB/#object-store-name
+  final storeName = 'analytics_cached_events';
+
+  late final Future<void> _databaseOpenEvent = _openDatabase();
+
+  late final IDBDatabase _database;
+
+  /// Checks for IDB availability and attempts to open the database.
+  Future<void> _openDatabase() async {
+    if (indexedDB == null) {
+      throw const InvalidStateException('IndexedDB is not available');
+    }
+    final openRequest = indexedDB!.open(databaseName, 1)
```

Comment:
Is `indexedDB` a nullable property? if so
```suggestion
    final db = indexedDB;
    if (db == null) {
      throw const InvalidStateException('IndexedDB is not available');
    }
    final openRequest = db.open(databaseName, 1)
```